collector/src/compile/execute/mod.rs

```diff
@@ -636,12 +650,27 @@ fn parse_self_profile(
     let (profile, files) = if let Some(profile_path) = full_path {
         // measureme 0.8+ uses a single file
         let data = fs::read(&profile_path)?;
-        let results = analyzeme::ProfilingData::from_paged_buffer(data, None)
-            .map_err(|error| {
-                eprintln!("Cannot read self-profile data: {error:?}");
-                std::io::Error::new(ErrorKind::InvalidData, error)
-            })?
-            .perform_analysis();
+
+        // HACK: `decodeme` can unexpectedly panic on invalid data produced by rustc. We catch this
+        // here until it's fixed and emits a proper error.
+        let res =
+            std::panic::catch_unwind(|| analyzeme::ProfilingData::from_paged_buffer(data, None));
+        let results = match res {
+            Ok(Ok(profiling_data)) => profiling_data.perform_analysis(),
```

Comment:
Does this also need to be in the catch_unwind?

Reply:
My understanding was that from the buffers being read (+ validation), then the data should be in a better shape, but I don't know for sure. I haven't checked deeply into the entire call stack there as the panics were really at the start of the entire process.